src/main/scala/org/broadinstitute/dsde/snoop/ws/SnoopApiService.scala

```diff
@@ -49,7 +88,29 @@ trait SnoopApiService extends HttpService {
       get {
         requestContext => requestContext.complete(requestContext.request.headers.mkString(",\n"))
       }
-    } ~
+    }
+}
+
+// this trait defines our service behavior independently from the service actor
+@Api(value = "workflowExecutions", description = "Snoop Workflow Execution API", position = 1)
+trait WorkflowExecutionApiService extends HttpService {
+  import WorkflowExecutionJsonSupport._
+  import SprayJsonSupport._
+
+  def executionServiceHandler: RequestContext => WorkflowExecutionService
+
+  def workflowRoutes = startWorkflowRoute ~ workflowStatusRoute
```

Comment:
does it work if you put the cookie directive up here? That way you don't need to duplicate the cookie extraction.